packages/network-controller/src/NetworkController.ts

```diff
@@ -3,6 +3,7 @@ import Subprovider from 'web3-provider-engine/subproviders/provider';
 import createInfuraProvider from 'eth-json-rpc-infura/src/createProvider';
 import createMetamaskProvider from 'web3-provider-engine/zero';
 import { Mutex } from 'async-mutex';
+import { v1 as random } from 'uuid';
```

Comment:
Did you consider using v4 instead? That's typically what I've seen us use elsewhere for random IDs.

Reply:
Oh. Yeah that makes sense now looking at their docs. I'm seeing many uses of v1 in this repo but hadn't seen any instances of v4. But looks like v4 is more suited to this particular use.